### PR TITLE
Use retranslate for translations and simplify config

### PR DESCRIPTION
**Pull request by oliverviljamaa**
# v2.0.0
## Use retranslate for translations and simplify config

*Breaking changes*:
* `PublicNavigation` is now a named export, not a default one (in preparation for new components)
* `retranslate` is expected as a peer dependency from consumers

### CODE REVIEW

File: src/Locale.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  PL: 'pl',
+  PT: 'pt',
+  RO: 'ro',
+  RU: 'ru',
+  SG: 'sg',
+  TR: 'tr',
+  US: 'us',
+};
+
+const LocaleValues = Object.keys(Locale).map(key => Locale[key]);
```
Comment (coolsebz) [inline src/Locale.js]: Object.values?

Comment (coolsebz) [reply src/Locale.js]: as we discussed, this was done on purpose and we're avoiding introducing another polyfill so just running with this implementation

File: src/PublicNavigation/items/l10n/interpolate.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
 export default function interpolateLinkForLocale(link, locale) {
-  return link.replace('{{localePath}}', locale);
+  return link.replace(':locale', locale);
```
Comment (matallo) [inline src/PublicNavigation/items/l10n/interpolate.js]: is this safe enough? When looking for other i18n stuff I encountered this library https://airbnb.io/polyglot.js/ and I think is a good source of inspiration

doc https://github.com/airbnb/polyglot.js#interpolation
source https://github.com/airbnb/polyglot.js/blob/master/index.js

Of course when I originally suggested this was in a rush and didn't give a proper thought

File: src/Locale.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const Locale = {
```
Comment (matallo) [inline src/Locale.js]: shall we go with the extended locales (with language + country) and methods to get each part?

File: src/PublicNavigation/items/l10n/interpolate.js (hunk 2 above)
Comment (oliverviljamaa) [reply src/PublicNavigation/items/l10n/interpolate.js]: What do you see as the danger? The same repo controlling the interpolate function also controls the link content (they don't even go through translations), so I'm not sure whether safety is an issue here.

File: src/Locale.js (hunk 3 above)
Comment (oliverviljamaa) [reply src/Locale.js]: Hmm, could you elaborate what we could do here?

File: src/PublicNavigation/items/l10n/interpolate.js (hunk 2 above)
Comment (matallo) [reply src/PublicNavigation/items/l10n/interpolate.js]: I was not referring that much regarding security but the possibility of unintended consequences with the regex. As you say the same repo controlling the interpolate function also controls the link content so it was just nitpicking than another thing

File: src/Locale.js (hunk 3 above)
Comment (matallo) [reply src/Locale.js]: sure, i.e.:

```
  au: 'en-AU',
  br: 'pt-BR',
  ca: 'en-CA',
  ch: 'de-CH',
  de: 'de-DE',
  es: 'es-ES',
...
```

Comment (oliverviljamaa) [reply src/Locale.js]: Doesn't seem to be worth it at the moment, as we're working with language and locale separately anyway. However, this logic needs to reside in frontend-service anyway!

**Pull request merged**